---
name: BMEG3105-Fall-2022
title: null
title-heading: false
layout: default
---


## BMEG3105: Data analytics for personalized genomics and precision medicine-Fall 2022

[<a href="https://forms.gle/RjKZaLafeDtEus2p9">Pre-course survey</a>, <a href="http://piazza.com/cuhk.edu.hk/fall2022/bmeg3105">Piazza</a>, 
<a href="https://docs.google.com/spreadsheets/d/1lA_eLCQawKLG2IWNPJ3gDa9bKp9adOZcQMdb_ciLgEg/edit?usp=sharing">Scribing preference</a>,
<a href="#logistics">Logistics</a>, <a href="#schedule_materials">Course schedule and materials</a>, <a href="#assignments">Assignments</a>]

### <a>Course description</a>
With social-economic development, people are increasingly caring about health. Consequently, in the field of genomics and healthcare, especially personalized genomics and precision medicine, we have accumulated a tremendous amount of data, which are waiting to be analyzed. This course is designed to equip students with the ability to analyze such data, which would benefit both the students' personal development and society. In the course, we will cover high-throughput experimental methods, standard data processing pipelines, sequence alignment and mapping, foundational concepts of data analytics, data exploration and visualization, clustering and classification, dimension reduction, and their applications in personalized genomics and precision medicine. 
For personalized genomics, we will also cover the integration of heterogeneous sequencing and non-sequencing data, single-cell data analysis, multi-omics analysis methods, and cancer genomics.
For precision medicine, we will cover protein-RNA interactions, biological graph analysis, and a gentle introduction to biomedical imaging and electronic health records.

#### Teaching team
Lecturer: 
Yu LI (<span style="color: #0099e6">liyuATcse.cuhk.edu.hk</span>), SHB-106. Office hour: 3pm-5pm, Friday <br>
TA: <br>
Qinze YU (<span style="color: #0099e6">qzyu22ATcse.cuhk.edu.hk</span>), SHB-116. Office hour: 2pm-4pm, Monday <br>
Yixuan WANG (<span style="color: #0099e6">yxwangATcse.cuhk.edu.hk</span>), SHB-116. Office hour: 2pm-4pm, Tuesday <br>
Liang HONG (<span style="color: #0099e6">lhong22ATcse.cuhk.edu.hk</span>), SHB-116. Office hour: 1pm-3pm, Monday <br>

#### Time and location
<b>Wednesday</b>: <b>9:30am-11am</b>, <b>SC-L4</b>. <br>
<b>Friday</b>: <b>9:30am-10:15am</b>, <b>LSB-LT3</b>. <br><b>
Friday</b>: <b>10:30-11:15am</b>, <b>LSB-LT3</b>. Tutorial <br>

#### Format
Mixed. Slides will be available the day before the lecture day. Video recordings will be available after the lecture.

### <a id="logistics">Logistics</a>



#### Communications
<b>Blackboard</b> is the main software to manage the course, and grading will be through blackboard. We will use <b>Piazza</b> (<a href="http://piazza.com/cuhk.edu.hk/fall2022/bmeg3105">BMEG3105</a>) for discussion. You can ask questions through Piazza, even anonymously. For a personal matter, please use the private post to the instructor and the TA. You are also very welcomed to send <b>emails</b> to the instructor and TAs.


#### Grading
<ul>
<li><b>Homework (20%)</b>: Three grading homework (5%+5%+5%) and one non-grading programming assignment (5%, make sure you learn something from it).</li>
<li><b>Scribing (10%)</b>: Grading scribing. Summarize one of the lectures. Submit it within one week after the course. Each student should do at least one lecture. You can sign for at most two, for additional 1%.</li>
<li><b>In-class quiz (10%)</b>: Two in-class quizzes. The questions will be simple. Mainly for checking the participation. The exact date are in the below <a href="#schedule_materials">schedule</a>.</li>
<li><b>Midterm (20%)</b>: A grading midterm exam. One bonus question (2%). </li>
<li><b>Project (20%)</b>: A grading project. You can give us your project and seek our help or we will predefine some projects for you to choose (You should submit a mid-term report (5%), a final report (7%) + presentation (3%) together with the implementation (5%).) </li>
<li><b>Final (20%)</b>: A grading finals.</li>
</ul>

<b>Bonus (up to 6%)</b>: One bonus question in Midterm. One additional scribing: 1%. <b>Pre-course survey</b> + <b>Post-lecture survey</b>: 0.5% for each, and the maximum is 3%. I do encourage you to complete all of them so that to let me know your feedback and adjust the course accordingly. Send your names to the TA. 

#### Open-book quiz and exam policy
All exams and quiz are open-book. You are allowed to take any <b>paper-based materials</b>. However, no phone or computer is allowed. Other communication tools are also not allowed. Discussion is not allowed.


#### Programming
Python (the TA will prepare a recitation class to introduce it, mainly for the non-grading homework and your project) or any other you are familiar with. For python, we suggest you to use <a href="https://colab.research.google.com/notebooks/intro.ipynb">Colab</a>. <br>
The programming credits include Non-grading assignment (10%) and Grading programming included in the project (5%). The bonus is sufficient to cover all the programming credit. If you really do not want to try hand-on experiments at all. We do encourage you try.

#### Scribing
Please sign <a href="https://docs.google.com/spreadsheets/d/1lA_eLCQawKLG2IWNPJ3gDa9bKp9adOZcQMdb_ciLgEg/edit?usp=sharing">Scribing preference</a>. We should have at least one student for each lecture. We may adjust the assignment if necessary. Notice that your note and scribing will be posted online, for others reference. You can choose to remove your name or not. Deadline for signing the scribing: **<span style="color:red;">11:59 pm on 12th Sep</span>**. After that, the Google sheet will be closed.

#### Projects
We will do the project individually. You can give us your project and seek our help or we will predefine some projects for you to choose. Some potential projects:
<ul>
<li>From reads to gene expression matrix processing pipeline.</li>
<li>Gene expression matrix processing pipeline.</li>
<li>Single-cell RNA-seq processing pipeline.</li>
<li>Bio-image classification. </li>
<li>Cancer gene identification. </li>
<li>Gene enrichment analysis.</li>
</ul>
Both the mid-term report (1 page) and the final report should be submitted.

#### Late days
Each student will have <b>6 late days</b> to turn in assignments, which can be used on A1, A2, A3, PA1, and the project mid-term report. They cannot be used on the project final report and the scribing note. A maximum of 2 late days can be used for each assignment. Grades will be deducted by 25% for each additional late day. 

#### Post-lecture survey 
Deadline for each survey: **<span style="color:red;">11:59pm on the day before the next lecture</span>**. We do this because I could have time to answer the questions you mentioned in the survey. Please fill 1 in the Google sheet: <a href="https://docs.google.com/spreadsheets/d/1D6fW_VH6zv6gjejbQ6u5PGJIWbgWpFe-sDG-TuiFnsI/edit?usp=sharing">Survey results</a>, once you have finished one survey. Usually, we will trust the 1s you fill in the Google sheet. But we will check the things in detail if the number of survey forms we received and the number of 1s on the Google sheet is not consistent.


### <a id="schedule_materials">Course schedule and materials</a>



| Lecture | Date  |Location| Topic | Slides/Video | Notes | Reading | Important dates (All due at **<span style="color:red;">11:59 pm</span>**)|
| ------- |------- |-------| ------| -------|-------|---------|----------------|
|1| Sep 7 (Wed) | SC-L4 | Introduction | <a href="https://www.dropbox.com/s/6xg1p8gccltj5sd/lec1-Intro.pdf?dl=0">Lec-1</a>    | [note1](./notes/BMEG3105_Lec01_s01.pdf) | <a href="https://www.dropbox.com/s/n5p1prup7htg5zw/BMEG3150_Course%20Outline_1st%20Term_2022-23.pdf?dl=0">Course outline</a>  |  |
| 2       | Sep 9 (Fri) | LSB-LT3 | Data & Python   | <a href="https://www.dropbox.com/s/tfzo1ttzb5fxjgk/lec2-data%20and%20python.pdf?dl=0">Lec-2</a>  | [note1](./notes/BMEG3105_Lec02_s01.pdf),  [note2](./notes/BMEG3105_Lec02_s02.pdf),  [note3](./notes/BMEG3105_Lec02_s03.pdf),  [note4](./notes/BMEG3105_Lec02_s04.pdf) | [sample code](https://colab.research.google.com/drive/1cdbEwWRqw2QUtITqvBj7u5m8NRgnY89Y?usp=sharing), [python cheat sheet](./reading/python_cheat_sheet.pdf), [numpy cheat sheet](./reading/numpy_cheat_sheet.pdf) | <a href="#assignments">PA 0</a> posted|
| 3       | Sep 14 (Wed) | SC-L4 | Sequence and DP | <a href="https://www.dropbox.com/s/s6gdienq869y3qq/lec3-DP.pdf?dl=0">Lec-3</a> | [note1](./notes/BMEG3105_Lec03_s01.pdf), [note2](./notes/BMEG3105_Lec03_s02.pdf), [note3](./notes/BMEG3105_Lec03_s03.pdf) | <a href="https://www.dropbox.com/s/tdxlt6yaigvj6jy/chapter2%263.pdf?dl=0">Chapter 2&3</a>        | **<span style="color:red;">PA0 due</span>**|
| 4       | Sep 16 (Fri) | LSB-LT3 | Assembly & Mapping| <a href="https://www.dropbox.com/s/dee1kc48dihcjck/lec4-Assembly%20and%20mapping.pdf?dl=0">Lec-4</a>  | [note1](./notes/BMEG3105_Lec04_s01.pdf), [note2](./notes/BMEG3105_Lec04_s02.pdf) |  <a href="https://colab.research.google.com/drive/1QmT6a7XumAYbd_9NhVfLuu9VvpR7-mtP?usp=sharing">Sample code</a>, <a href="https://www.dropbox.com/s/j6gu44xszr9e8jk/A%20survey%20of%20best%20practices%20for%20RNA-seq%20data%20analysis.pdf?dl=0">RNA-seq analysis</a>, [intro to python tutorial](./reading/Tutorial_1_intro%20to%20python.pptx), [anaconda starter guide](./reading/Anaconda-Starter-Guide.pdf), [conda cheatsheet](./reading/conda-cheatsheet.pdf)| A1 posted|
|5| Sep 21 (Wed) | SC-L4 |Data exploration | <a href="https://www.dropbox.com/s/485rsvfrb3ohws0/lec5-Exploration%20and%20cleaning.pdf?dl=0">Lec-5</a>      | [note1](./notes/BMEG3105_Lec05_s01.pdf) |<a href="https://github.com/chenomg/CS_BOOKS/blob/master/Python%20for%20Data%20Analysis%2C%202nd%20Edition.pdf">Python for DA</a>, <a href="https://colab.research.google.com/drive/1z0_IEP-tOZgt7auZqEMtLvmafVuHRP0d?usp=sharing">Sample code</a>, <a href="https://colab.research.google.com/drive/1p22oCIaFFfDU9BwzwdBVq3KPvD4BqBMS?usp=sharing">Sample code-2</a>         |  |
| 6      | Sep 23 (Fri) | LSB-LT3 | Distance and clustering | <a href="https://www.dropbox.com/s/2b4w8xsb4culrh5/lec6-Distance%20and%20clustering.pdf?dl=0">Lec-6</a> | [note1](./notes/BMEG3105_Lec06_s01.pdf)  | <a href="https://github.com/FaizSaeed/Data-Science-Course/blob/master/Book/Introduction%20to%20Data%20Mining_Pang%20Ning%20Tan.pdf">Data mining book</a>,  <a href="https://colab.research.google.com/drive/1z0_IEP-tOZgt7auZqEMtLvmafVuHRP0d?usp=sharing">Sample code</a>, <a href="https://colab.research.google.com/drive/1p22oCIaFFfDU9BwzwdBVq3KPvD4BqBMS?usp=sharing">Sample code-2</a>, [intro to pandas&numpy tutorial](./reading/BMEG3105-Introduction%20to%20Pandas%20and%20NumPy.pptx) | |
| 7      | Sep 28 (Wed) | SC-L4 | Classification  | <a href="https://www.dropbox.com/s/9uuygvv2xckow0n/lec7-Classification.pdf?dl=0">Lec-7</a>  |  [note1](./notes/BMEG3105_Lec07_s01.pdf), [note2](./notes/BMEG3105_Lec07_s02.pdf)     | <a href="https://github.com/FaizSaeed/Data-Science-Course/blob/master/Book/Introduction%20to%20Data%20Mining_Pang%20Ning%20Tan.pdf">Data mining book</a>, <a href="https://colab.research.google.com/drive/1ogX-QiR1jBWuZjbTXGFhi9H-0XdpGNhU?usp=sharing">Correlation</a>  | **<span style="color:red;">A1 due</span>**|
|8| Sep 30 (Fri) | LSB-LT3 | Classification | <a href="https://www.dropbox.com/s/nmis4w3kocq57an/lec8-Classification.pdf?dl=0">Lec-8</a>  |[note1](./notes/BMEG3105_Lec08_s01.pdf), [note2](./notes/BMEG3105_Lec08_s02.pdf) | <a href="https://github.com/FaizSaeed/Data-Science-Course/blob/master/Book/Introduction%20to%20Data%20Mining_Pang%20Ning%20Tan.pdf">Data mining book</a>, [tut4 slides](reading/bmeg3105_tut4.pptx) |  |
| 9      | Oct 5 (Wed) | SC-L4 | Perf evaluation  | <a href="https://www.dropbox.com/s/bnk9ag2y3dll8ac/lec9-Performance%20evaluation.pdf?dl=0">Lec-9</a>  |   [note1](./notes/BMEG3105_Lec09_s01.pdf), [note2](./notes/BMEG3105_Lec09_s02.pdf), [note3](./notes/BMEG3105_Lec09_s03.pdf)    | <a href="https://github.com/FaizSaeed/Data-Science-Course/blob/master/Book/Introduction%20to%20Data%20Mining_Pang%20Ning%20Tan.pdf">Data mining book</a>  | A2 posted|
| 10     | Oct 7 (Fri) | LSB-LT3 | Feat selection | <a href="https://www.dropbox.com/s/x09vzfs58d5otk3/lec10-Feature%20selection%20and%20DR.pdf?dl=0">Lec-10</a>  | [note1](./notes/BMEG3105_Lec10_s01.pdf), [note2](./notes/BMEG3105_Lec10_s02.pdf)      | <a href="https://github.com/probml/pml-book">PML book</a>, [tut5 slides](./reading/Tutorial_5_model%20evaluation.pptx)       |[note1](./notes/BMEG3105_Lec11_s01.pdf), [note2](./notes/BMEG3105_Lec11_s02.pdf) |
| 11     | Oct 12 (Wed) | SC-L4 | Dim reduction | <a href="https://www.dropbox.com/s/h14zcm4xb1xzd4e/lec11-DR%20and%20NN.pdf?dl=0">Lec-11</a>   | | <a href="https://github.com/probml/pml-book">PML book</a> |  |
| 12      | Oct 14 (Fri) | LSB-LT3 | Overfitting | <a href="https://www.dropbox.com/s/8e6y3u8ccp8a9g1/lec12-Overfitting.pdf?dl=0">Lec-12</a> | [note1](./notes/BMEG3105_Lec12_s01.pdf)|  <a href="https://d2l.ai">D2L book</a>,   [tut6 slides](./reading/TUT6.pptx)    |   |
| 13    | Oct 19 (Wed) | SC-L4 | Mid-term review | <a href="https://www.dropbox.com/s/mja4931lubfrxau/lec13-Mid-term%20review.pdf?dl=0">Lec-13</a>  |       | [tut A2 (passcode: ef94@^+f)](https://cuhk.zoom.us/rec/share/7mdR-4dzVqp8F-w3bQguypUuB7YeicQwQdSxzbBSaxl_5aHMsZ0oyKqp5HchoNzG.RlEQnIiMVNMvjtKl <br/>Passcode: ef94@^+f) | **<span style="color:red;">ParticipationQ,</span>** **<span style="color:red;">A2 due</span>**|
| 14      | Oct 21 (Fri) | LSB-LT3 | Mid-term | |- |- |**<span style="color:red;">8:30am-11:15am, Mid-term</span>**|
||||<span style="color:blue;">Module 2 start</span>|||||
| 15      | Oct 26 (Wed) | SC-L4 |  Multi-omics overview  | <a href="https://www.dropbox.com/s/y41ahspm9vg2xjx/lec15-multi-omics.pdf?dl=0">Lec-15</a> |   [note1](./notes/BMEG3105_Lec15_s01.pdf)   |<a href="https://www.cancer.gov/about-cancer/understanding/what-is-cancer">Intro to cancer</a>, <a href="https://www.ebi.ac.uk/training/materials/cancer-genomics-materials/">Cancer genomics</a> | PA1 posted|
| 16      | Oct 28 (Fri) | LSB-LT3 | Cancer genomics overview | <a href="https://www.dropbox.com/s/iqlqv2q8us9bmtk/lec16-cancer%20genomics.pdf?dl=0">Lec-16</a>  | [note1](./notes/BMEG3105_Lec16_s01.pdf), [note2](./notes/BMEG3105_Lec16_s02.pdf)      |<a href="https://www.cancer.gov/about-cancer/understanding/what-is-cancer">Intro to cancer</a>, <a href="https://www.ebi.ac.uk/training/materials/cancer-genomics-materials/">Cancer genomics</a> |  |
| 17      | Nov 2 (Wed) | SC-L4 | Genomics data analysis | <a href="https://www.dropbox.com/s/4sqm4z0z8qbm96i/lec17-genomics%20analysis.pdf?dl=0">Lec-17</a> |  | <a href="https://www.ebi.ac.uk/training/materials/cancer-genomics-materials/">Cancer genomics</a>, <a href="https://drive.google.com/drive/folders/1y7q0gJ-ohNDhKG85UTRTwW1Jkq4HJ5M3">GATK</a>, <a href="https://www.youtube.com/watch?v=xw419NKqMqw">GWAS</a>, <a href="https://www.youtube.com/watch?v=IAu44BkOaSs">Epigenetics</a>, <a href="https://www.encodeproject.org/atac-seq/">ENCODE</a>|  |
| 18      | Nov 4 (Fri) | LSB-LT3 | Single cell genomics | <a href="https://www.dropbox.com/s/hwz4z0dnamom81c/lec18-single%20cell.pdf?dl=0">Lec-18</a>   |       | <a href="https://www.dropbox.com/s/7kqrm7jdn4swsor/Single%20cell-Current%20best%20practice.pdf?dl=0">Current best practice</a>, <a href="https://www.dropbox.com/s/ibcw0mlsq4f2zl2/Single%20cell-Tutorial.pdf?dl=0">Tutorial-1</a>, <a href="https://broadinstitute.github.io/2019_scWorkshop/">Tutorial-2</a>, <a href="https://www.singlecellcourse.org/index.html">Tutorial-3</a>, <a href="https://www.dropbox.com/s/99o7ph37b9uveau/Single%20cell-Clustering%20challenges.pdf?dl=0">Clustering challenges</a> , <a href='https://colab.research.google.com/drive/1wdz8uKWFafey16DU3pCPdvM8FOAGuMGt?usp=share_link'>PyTorch Tutorial</a>, [GATK-related software installation ](./reading/TUT9.pptx) | |
| 19      | Nov 9 (Wed) | SC-L4 | Data visualization |  |       |         | **<span style="color:red;">Project M-report</span>**|
| 20     | Nov 11 (Fri) | LSB-LT3 | Protein-RNA     |  |       |         ||
| |||<span style="color:blue">Module 3 start</span>|||||
| 21     | Nov 16 (Wed) | SC-L4 | Deep learning/CNN  |  |       |         | **<span style="color:red;">PA1 due</span>**, A3 posted|
| 22      | Nov 18 (Fri) | LSB-LT3 | How to present |  |       |         ||
| 23      | Nov 23 (Wed) | SC-L4 | EHRs & Bio-graph |  |       |         ||
| 24      | Nov 25 (Fri) | LSB-LT3 | Project pres   |  | | | **<span style="color:red;">A3 due</span>**|
| 25      | Nov 30 (Wed) | SC-L4 | Course review |  | | | **<span style="color:red;">ParticipationQ</span>**|
| 26      | Dec 2 (Fri) | LSB-LT3 | Project pres  | | | | **<span style="color:red;">Project report</span>**|

### <a id="assignments">Assignments</a>
<ul>
<li>Programming Assignment 0: Get yourself familiar with <a href="https://colab.research.google.com/notebooks/intro.ipynb">Colab</a>, set up the environment and run this <a href="https://colab.research.google.com/drive/1cdbEwWRqw2QUtITqvBj7u5m8NRgnY89Y?usp=sharing
">code</a>. </li>

<li>Assignment 1: Basic concept of data analytics-1</li>

<li>Assignment 2: Basic concept of data analytics-2</li>

<li>Programming Assignment 1: Application of DA to biology</li>

<li>Assignment 3: DA in Personalized Genomics and Precision Medicine</li>

</ul>
